examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -226,31 +227,31 @@ function default_remaining_tendency!(Yₜ, Y, p, t)
     @. ᶜK = norm_sqr(ᶜuvw) / 2
 
     # Mass conservation
-
-    @. Yₜ.c.ρ -= divₕ(ᶜρ * ᶜuvw)
+    FT = eltype(Yₜ.c.ρ)
+    @. Yₜ.c.ρ -= split_divₕ(ᶜρ * ᶜuvw, FT(1))
```

Comment:
```suggestion
    @. Yₜ.c.ρ -= split_divₕ(ᶜρ * ᶜuvw, 1)
```
This operator should be able to automatically promote integers to floating-point values.

Based on past experiments, specifying `FT` conversions in broadcast expressions incurs a higher latency penalty than performing automatic conversions when evaluating each point. While automatic conversions are easy for the compiler to inline, explicit conversions are much harder because they are passed through the entire broadcasting API, and sometimes they aren't inlined at all (incurring a slight runtime penalty in addition to the latency penalty).